### PR TITLE
Modernise the 'New view' and 'New node' pages

### PR DESCRIPTION
**Pull request by janfaracik**
## What's changed?

This PR updates the appearance of several (not all!) form controls to give Jenkins a more modern, consistent appearance. The main focus of this PR is to modernise the 'New view' and 'New node' pages, however, some of the changes, such as the change in `input` design will ripple throughout Jenkins.

The end goal would be to update all the form controls across Jenkins, ensuring consistency, modernity and better accessibility throughout the service.

## Test this pull request

```command
docker run --rm -ti -p 8080:8080 -e ID=5842 -e MERGE_WITH=master jenkins/core-pr-tester
```

### Old
image

### New
image

### Old
image

### New
https://user-images.githubusercontent.com/43062514/138265559-13998e0d-421d-4a93-81c0-d4a37a7a7260.mov

### Proposed changelog entries

* Modernise the 'New view' and 'New node' pages.

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by fqueiruga — COMMENTED** [review]
I think this is a GREAT PR. Congratulations!!!!! 🎉 I do have a few comments if you don't mind:

- 350px inputs are way too short for. Going for ~700px/43rem is always a solid choice.
- I think you could very well go all in with the configuration. Stuff like padding, label font weight, box shadows, etc.
- Border radius should IMO be 4px to be consistent with the rest of the UI and plugins.

My only -1 would be having custom buttons. I don't think there's any need for that. Existing buttons are widely used and won't probably be changed. And if they need changes, it should be global IMO. This is coming from my personal experience with the buttons revamp last year.

Also, have you tried applying these styles to the `.setting-input` class, just to see what happens? Any long term form strategy to update inputs would be most effective by acting on that class.

**Review by timja — COMMENTED** [review]
Generally I think this is really good, my main concern is that some pages are going to look weird because the changes only apply to text input currently

Examples where it looks weird:

<details>
<summary>Password input</summary>

image

</details>

<details>
<summary>Number / dropdowns</summary>

image


</details>

<details>
<summary>Configure system page</summary>

image

</details>

**Review by uhafner — COMMENTED** [review]
I think the design looks great! Since we have so many different places where forms are used this PR might need some more time until we can ensure that we do not break too much forms?

**Review by timja — APPROVED** [review]
Last weirdness is generally just things that haven't been constrained by the max-width, mostly advanced buttons but that will be fixed in a follow-up.

I've fixed the acceptance test harness issues that are causing this build to fail, but that caused lots of other ones to fail which i'm fixing as I get time, (so I haven't released the tests yet).

They are failing because the radio is no longer a real radio button (hidden by opacity: 0) so we have to click the label, the tests have this thing where they navigate by `xpath` and the label doesn't have an xpath so the code needs adjusting to find the input from the label, (which is trivial, it's just finding all the places), and some places need updating to use label text instead of the input name

**Review by fqueiruga — CHANGES_REQUESTED** [review]
I have some, I think critical, change requests on JS code that I think will break backwards compatibility. Should be an straightforward fix. After that, it's an immediate ✅ from me.

I do suggest flattening out SCSS selectors, especially class definitions done by nesting. These are in my experience very difficult to maintain, especially for non frontend-specialised people.

This is an amazing PR and the forms look so much better live. I also expected the mixed form styles to be way more jarring but it's not that bad for an intermediate step. Cheers!

**Review by uhafner — COMMENTED** [review]
I general: a big improvement 🎉. Some details are still broken, I'm not sure if that should be fixed before we merge this. One thing that I don't like is the fixed size of the forms which looks kind of odd on my 27'' monitor. But maybe this is a bug because some elements are still on the right side.

Some details I found (description is always after the screenshot).

Bildschirmfoto 2021-10-27 um 11 16 15

- The section headers are not always correct (orange rectangle).
- The gap after the header text is too small (red rectangle). 

Bildschirmfoto 2021-10-27 um 11 11 28

- Some elements (buttons) float to the right (orange rectangle)
- All blocks do not fill the available space anymore (red rectangle). From the discussion this seems to be intended, but why is the background area still so wide? For people with a monitor most part of the screen is now empty (and large descriptions fill a couple of lines)
- Select form element is not yet styled correctly
- Autocompletion text field is not yet styled correctly

Bildschirmfoto 2021-10-27 um 11 29 29

- Custom entries are not yet styled correctly (this can be implemented in the affected plugins step by step)

Bildschirmfoto 2021-10-27 um 11 39 25

- In the pipeline step snippet generator somehow the fonts of descriptions are smaller

**Review by fqueiruga — APPROVED** [review]
@timja it's a go from me then :D congrats @janfaracik

**Review by uhafner — APPROVED** [review]
Fine for me as well now. Thanks for your work!

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Comment by timja** [comment]
Some minor test fixups needed:
`lib.form.AdvancedButtonTest.testSectionInsideOfAdvanced`
`lib.form.AdvancedButtonTest.testNestedOptionalBlock`

**Comment by janfaracik** [comment]
> Some minor test fixups needed:
> `lib.form.AdvancedButtonTest.testSectionInsideOfAdvanced`
> `lib.form.AdvancedButtonTest.testNestedOptionalBlock`

Fixed now :)

**Review by oleg-nenashev — APPROVED** [review]
Looks good to me. IIRC I tested it locally a few weeks ago and new layouts looked really well

**Comment by oleg-nenashev** [comment]
@timja I suggest merging it into the weekly. the 24hrs timeout expires just one hour after the release should start IIUC

**Comment by timja** [comment]
> @timja I suggest merging it into the weekly. the 24hrs timeout expires just one hour after the release should start IIUC

There’s a security release on Thursday, delaying bigger merges till after that

**Comment by daniel-beck** [comment]
[JENKINS-67071](https://issues.jenkins.io/browse/JENKINS-67071) suggests this causes a regression in job config forms.

**Comment by timja** [comment]
> [JENKINS-67071](https://issues.jenkins.io/browse/JENKINS-67071) suggests this causes a regression in job config forms.

Addressed in https://github.com/jenkinsci/jenkins/pull/5893

**Comment by timja** [comment]
Spacing is off:

image

Validation icon touched the input

**Comment by timja** [comment]
Disabled project text and icon is mis-aligned:

image

**Comment by timja** [comment]
Help icon broken for freestyle build Set build status pending github commit (GitHub plugin)

image

(assume it was this PR)

**Comment by basil** [comment]
After pulling in this commit, I can no longer add a Maven installation via the UI (see [JENKINS-67109](https://issues.jenkins.io/browse/JENKINS-67109)).

**Comment by basil** [comment]
@janfaracik This PR breaks the following tests and blocks weekly updates to the BOM (cf. jenkinsci/bom#717 and jenkinsci/bom#719):

- `hudson.plugins.throttleconcurrents.ThrottleJobPropertyTest#clearConfiguredCategories`
- `hudson.plugins.throttleconcurrents.ThrottleQueueTaskDispatcherTest#testShouldConsiderTaskAsBlockableStillUponMatchingLabelPairWithLowestMax`
- `hudson.plugins.throttleconcurrents.ThrottleQueueTaskDispatcherTest#testShouldConsiderTaskAsBlockableStillUponMatchingMaxLabelPair`
- `hudson.plugins.throttleconcurrents.ThrottleQueueTaskDispatcherTest#testShouldConsiderTaskAsBlockableStillUponMatchingMaxLabelPairs`
- `hudson.plugins.throttleconcurrents.ThrottleQueueTaskDispatcherTest#testShouldConsiderTaskAsBuildableStillUponMismatchingMaxLabelPairs`
- `org.jenkinsci.plugins.pipeline.modeldefinition.StageInputTest.parametersInInput`

The tests will need to be adapted to this PR and new versions of the abovementioned plugins released.

**Comment by timja** [comment]
I have changed locally that fixes most of those, there’s one test i’m still debugging

**Comment by basil** [comment]
Also breaks `hudson.plugins.emailext.ExtendedEmailPublisherDescriptorTest#groovyClassPath`. Has an effort been made to systematically evaluate the impact of this PR and prepare plugins?

**Comment by uhafner** [comment]
I'm not sure if someone is still is working on the problematic width resolution change within this PR, I created an issue so this will not be forgotten: https://issues.jenkins.io/browse/JENKINS-67198

**Comment by NathanLindshield** [comment]
The screenshots above even show the fixed minimal width

**Comment by timja** [comment]
It's being addressed to a degree in https://github.com/jenkinsci/jenkins/pull/5923

There's additional work coming as well that was demoed in the previous UX sig meeting, around here:
https://youtu.be/SxSObgwjwt8?t=3428

Plenty of open PRs in this area:
https://github.com/jenkinsci/jenkins/pulls?q=is%3Apr+is%3Aopen+sort%3Aupdated-desc+author%3Ajanfaracik

Generally getting held up either adjusting tests to markup or fixing issues, but hopefully we'll get most of them merged in December

**Comment by NathanLindshield** [comment]
Thanks @timja, appreciate it! Great work!

**Comment by hankzhao-zp** [comment]
> I'm not sure if someone is still is working on the problematic width resolution change within this PR, I created an issue so this will not be forgotten: https://issues.jenkins.io/browse/JENKINS-67198

Is anyone looking at this question, it really bothers me

**Comment by janfaracik** [comment]
> > I'm not sure if someone is still is working on the problematic width resolution change within this PR, I created an issue so this will not be forgotten: https://issues.jenkins.io/browse/JENKINS-67198
> 
> Is anyone looking at this question, it really bothers me

I am :) It'll be updated as part of #5923, I've posted a WIP screenshot on the Jira thread if you're interested.

**Comment by basil** [comment]
Causes [JENKINS-68042](https://issues.jenkins.io/browse/JENKINS-68042)

File: core/src/main/resources/lib/form/helpLink.jelly
Comment (daniel-beck) [inline core/src/main/resources/lib/form/helpLink.jelly:57]: Caused [SECURITY-2779](https://www.jenkins.io/security/advisory/2022-06-22/#SECURITY-2781)

**Comment by daniel-beck** [comment]
Caused [JENKINS-69030](https://issues.jenkins.io/browse/JENKINS-69030)

**Comment by daniel-beck** [comment]
Contributed to [JENKINS-69549](https://issues.jenkins.io/browse/JENKINS-69549).

**Comment by daniel-beck** [comment]
Caused [JENKINS-72749](https://issues.jenkins.io/browse/JENKINS-72749)

**Pull request merged**
